keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ApproveOldVersionTests.kt

```diff
@@ -77,12 +78,18 @@ abstract class ApproveOldVersionTests<T : KeelRepository> : JUnit5Minutests {
       every { isImplicit() } returns true
       every { canPromote(any(), any(), any(), any()) } returns true
     }
```

Comment:
This test uses the sql database and tests that approving old versions works. All I had to do was add the new dependency. This gives me lots of confidence

Reply:
Are you doing anything to invoke that? Based on the appearance of the diff here it's not clear to me what difference in test behavior this makes.

Reply:
Right now we invoke the new code (the `EnvironmentConstraintRunner`) w/in the `EnvironmentPromotionChecker`. Because this code is using sql and not mocks it actually tests the new behavior more completely in my mind.